ruby/ql/lib/codeql/ruby/InclusionTests.qll

```diff
@@ -0,0 +1,133 @@
+/**
+ * Contains classes for recognizing array and string inclusion tests.
+ */
+
+private import ruby
+private import codeql.ruby.DataFlow
+private import codeql.ruby.controlflow.CfgNodes
+
+/**
+ * An expression that checks if an element is contained in an array
+ * or is a substring of another string.
+ *
+ * Examples:
+ * ```
+ * A.include?(B)
+ * A.index(B) !== -1
+ * A.index(B) >= 0
+ * ```
+ */
+class InclusionTest extends DataFlow::Node instanceof InclusionTest::Range {
```

Comment:
Why is this using data-flow nodes instead of CFG nodes?

Reply:
The JavaScript library does the same, so keeping it this way makes it easier to share the implementation of the query.